frontend/src/v5/ui/routes/dashboard/projects/containers/containersList/latestRevision/latestRevision.component.tsx

```diff
@@ -16,96 +16,34 @@
-import { Name, ProcessingStatus, QueuedStatus, Label, Container } from './latestRevision.styles';
+import { RevisionStatus } from '@/v5/ui/routes/dashboard/projects/containers/containersList/latestRevision/revisionStatus';
+import { IRevisionStatus } from '@/v5/ui/routes/dashboard/projects/containers/containersList/latestRevision/revisionStatus/revisionStatus.component';
+import { Label, Container } from './latestRevision.styles';
 
-interface ILatestRevision {
-	name: string;
-	status: ContainerStatuses;
-	error?: {
-		date: Date | null;
-		message: string;
-	}
+interface ILatestRevision extends IRevisionStatus {
+	hasRevisions: boolean;
 }
 
-export const LatestRevision = ({ name, status, error }: ILatestRevision): JSX.Element => (
+export const LatestRevision = ({ hasRevisions, ...props }: ILatestRevision): JSX.Element => (
 	<Container>
-		<Label>
-			<Trans
-				id="containers.list.item.latestRevision.label"
-				message="Latest revision: "
-			/>
-		</Label>
-
-		{(() => {
-			if (status === ContainerStatuses.QUEUED) {
-				return (
-					<QueuedStatus>
-						<Trans id="containers.list.item.latestRevision.status.queued" message="Queued" />
-					</QueuedStatus>
-				);
-			}
-
-			if (status === ContainerStatuses.PROCESSING) {
-				return (
-					<ProcessingStatus>
-						<Trans id="containers.list.item.latestRevision.status.processing" message="Processing" />
-					</ProcessingStatus>
-				);
-			}
-
-			if (status === ContainerStatuses.FAILED && error) {
-				return (
-					<>
-						<Name>
-							{name}
-						</Name>
-						<ErrorTooltip>
-							{error.date ? (
-								<Trans
-									id="containers.list.item.latestRevision.status.error.tooltipMessageWithDate"
-									message="The latest upload on <0>{date}</0> at <0>{time}</0> has failed due to <0>{message}</0>."
-									values={{
-										date: i18n.date(error.date),
-										time: i18n.date(error.date, {
-											hour: 'numeric',
-											minute: 'numeric',
-										}),
-										message: error.message,
-									}}
-									components={[
-										<b />,
-									]}
-								/>
-							) : (
-								<Trans
-									id="containers.list.item.latestRevision.status.error.tooltipMessageWithoutDate"
-									message="The latest upload has failed due to <0>{message}</0>."
-									values={{
-										message: error.message,
-									}}
-									components={[
-										<b />,
-									]}
-								/>
-							)}
-
-						</ErrorTooltip>
-					</>
-				);
-			}
-
-			return (
-				<TextOverflow>
-					<Name>
-						{name}
-					</Name>
-				</TextOverflow>
-			);
-		})()}
+		{!hasRevisions ? (
+			<Label>
+				<Trans
+					id="containers.list.item.latestRevision.emptyContainer"
+					message="Container empty"
+				/>
+			</Label>
+		) : (
+			<>
+				<Label>
+					<Trans
+						id="containers.list.item.latestRevision.label"
+						message="Latest revision: "
+					/>
+				</Label>
+				<RevisionStatus {...props} />
```

Comment:
@EmilTheSadCat 

I think this needs to be outside of the if statement? It's possible that we have no revision, but currently uploading one.